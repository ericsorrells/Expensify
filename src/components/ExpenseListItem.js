import React from 'react';
import { removeExpense } from '../actions/expenses'
import { connect } from 'react-redux';

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => ( 
  <div>
    <h3>Description: {description}</h3>
    <p>Amount: {amount} |
    Created: {createdAt}</p>
    <button onClick={() => {
      dispatch(removeExpense({ id }))
    }}
    >
      Remove Me
    </button>
  </div>
)

export default connect()(ExpenseListItem)