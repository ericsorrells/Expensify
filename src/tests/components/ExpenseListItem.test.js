import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';

const item = { 
  id: '1', 
  description: 'some description', 
  amount: 100, 
  createdAt: 0
}

test('<ExpenseListItem: renders correctly w props', () => {
  const component = shallow(<ExpenseListItem {...item} />);
  expect(component).toMatchSnapshot();
})