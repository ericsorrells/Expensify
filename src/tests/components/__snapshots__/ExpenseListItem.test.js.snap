// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpenseListItem: renders correctly w props 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Description: 
      some description
    </h3>
  </Link>
  <p>
    $1.00
    -
    December 31st, 1969
  </p>
</div>
`;
